tests/ui/resolve/export-fully-qualified.stderr

```diff
@@ -3,6 +3,11 @@ error[E0433]: failed to resolve: use of undeclared crate or module `foo`
    |
 LL |     pub fn bar() { foo::baz(); }
    |                    ^^^ use of undeclared crate or module `foo`
+   |
+help: consider importing this module
+   |
+LL +     use foo;
```

Comment:
This isn't that useful either 🤔

Reply:
There is a difference between `2015 edition` and later edition,

With this PR, the output for `2021 edition` is:
```console
rustc +dev2 --edition 2021 tests/ui/resolve/export-fully-qualified.rs
error[E0432]: unresolved import `foo`
 --> tests/ui/resolve/export-fully-qualified.rs:7:13
  |
7 |         use foo;
  |             ^^^ no external crate `foo`
  |
help: consider importing this module instead
  |
7 |         use crate::foo;
  |  
```

the testing framework is running without 2021, so it's suggesting to write this:
```rust
mod foo {
    pub fn bar() {
        use foo;
        foo::baz();
    } //~ ERROR failed to resolve: use of undeclared crate or module `foo`

    fn baz() { }
}
```

This is valid code for `2015 edition`. If we compile with a new edition we will suggest using `crate`:

```console
~/rust ❯❯❯ rustc +dev2 --edition 2018 tests/ui/resolve/export-fully-qualified.rs
error[E0432]: unresolved import `foo`
 --> tests/ui/resolve/export-fully-qualified.rs:7:13
  |
7 |         use foo;
  |             ^^^ no external crate `foo`
  |
help: consider importing this module instead
  |
7 |         use crate::foo;
  |             ~~~~~~~~~~

error: aborting due to previous error
```


Maybe it's related to this issue https://github.com/rust-lang/rust/issues/112924

Anyway, this suggestion seems right for different versions.

Or it's better to suggest writing like this?
```rust
crate::foo::baz();
```

Reply:
We should have two tests, one for 2015 and one for 2018, so that we can ensure that the suggestions keep being reasonable for both :)

Reply:
yes, added another test `tests/ui/resolve/export-fully-qualified-2018.rs`.